cpu/stm32/include/candev_stm32.h

```diff
@@ -37,13 +37,12 @@ extern "C" {
 
 #include "can/candev.h"
 
-#if defined(CPU_LINE_STM32F413xx) || defined(CPU_LINE_STM32F423xx)
+/** Number of channels in the device (up to 3) */
+#if defined(CAN3)
 #define CANDEV_STM32_CHAN_NUMOF 3
-#elif defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4)
+#elif defined(CAN2)
 #define CANDEV_STM32_CHAN_NUMOF 2
-#elif defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F3) || \
-      defined(CPU_FAM_STM32L4) || defined(CPU_LINE_STM32F722xx) || DOXYGEN
-/** Number of channels in the device (up to 3) */
+#elif defined(CAN1) || DOXYGEN
```

Comment:
I like the idea.
However, STM32 are not simply defined as CAN1, CAN2, etc..
It might be defined using something other than CANx. Depends on MCU series, Fx, Lx, Hx, etc..
i.e. STM32F0 defines CAN; STM32L4 defines CAN, CAN1(both share).

Perhaps it would be safe just using CPU family macro. unless we need to go through all the series(Fx, Hx, Lx) and ensure the logic works.

Reply:
It seems like there is either `CANx_BASE` or `CAN_BASE` - I think we can use that

Reply:
> It seems like there is either `CANx_BASE` or `CAN_BASE` - I think we can use that.
> 
> stm32g0 and stm32l5 have `SRAMCAN_BASE` - is this equivalent?

this is the RAM for shared message of FD CAN. FD CAN is backward-compatible with CAN 2.0(bxCan). I am not sure if this code works for FD CAN. It would be nice to have a board to test with it. It might not. 

BTW, 
for STM32F0 work, 
```suggestion
#elif defined(CAN) || defined(CAN1) || DOXYGEN
```